sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobBatch.java

```diff
@@ -0,0 +1,413 @@
+        this.batchMapping.put(contentId, batchOperationResponse);
+        return batchOperationResponse;
+    }
+
+    private BlobAsyncClientBase buildClient(String containerName, String blobName) {
+        return new BlobClientBuilder()
+            .endpoint(accountUrl)
+            .containerName(containerName)
+            .blobName(blobName)
+            .pipeline(batchPipeline)
+            .buildAsyncClient();
+    }
+
+    private BlobAsyncClientBase buildClient(String blobUrl) {
+        return new BlobClientBuilder()
+            .endpoint(blobUrl)
+            .pipeline(batchPipeline)
+            .buildAsyncClient();
+    }
+
+    private void sendCallback(HttpRequest request) {
+        int contentId = Integer.parseInt(request.getHeaders().remove(CONTENT_ID).getValue());
+
+        StringBuilder batchRequestBuilder = new StringBuilder();
+        appendWithNewline(batchRequestBuilder, "--" + batchBoundary);
+        appendWithNewline(batchRequestBuilder, CONTENT_TYPE);
+        appendWithNewline(batchRequestBuilder, CONTENT_TRANSFER_ENCODING);
+        appendWithNewline(batchRequestBuilder, String.format(CONTENT_ID_TEMPLATE, contentId));
+        batchRequestBuilder.append("\r\n");
+
+        String method = request.getHttpMethod().toString();
+        String urlPath = request.getUrl().getPath();
+        String urlQuery = request.getUrl().getQuery();
+        if (!ImplUtils.isNullOrEmpty(urlQuery)) {
+            urlPath = urlPath + "?" + urlQuery;
+        }
+        appendWithNewline(batchRequestBuilder, String.format(OPERATION_TEMPLATE, method, urlPath, HTTP_VERSION));
+
+        request.getHeaders().stream()
+            .filter(header -> !Constants.HeaderConstants.VERSION.equalsIgnoreCase(header.getName()))
+            .forEach(header -> appendWithNewline(batchRequestBuilder,
+                String.format(HEADER_TEMPLATE, header.getName(), header.getValue())));
+
+        batchRequestBuilder.append("\r\n");
+
+        batchRequest.add(ByteBuffer.wrap(batchRequestBuilder.toString().getBytes(StandardCharsets.UTF_8)));
+
+        this.batchMapping.get(contentId).setRequest(request);
+    }
+
+    private void appendWithNewline(StringBuilder stringBuilder, String value) {
+        stringBuilder.append(value).append("\r\n");
+    }
+
+    Flux<ByteBuffer> getBody() {
+        if (batchOperationQueue.isEmpty()) {
+            throw logger.logExceptionAsError(new UnsupportedOperationException("Empty batch requests aren't allowed."));
+        }
+
+        while (!batchOperationQueue.isEmpty()) {
+            Mono<? extends Response> batchOperation = batchOperationQueue.pop();
+            Disposable disposable = batchOperation.subscribe();
+            while (!disposable.isDisposed()) {
+                // Wait until the batch operation has processed in the pipeline.
+                // This is used as opposed to block as it won't trigger an exception if ran in a Reactor thread.
+            }
+        }
```

Comment:
Is it imperative that each operation complete before the next starts? Can you run them concurrently with a flatMap / collect style approach. The current code goes into a very tight while loop which isn't ideal.

Reply:
Will try out a Flux flatMapping that has a single tight loop instead of this numerous tight loop implementation.